sway-lib-std/src/storage.sw

```diff
@@ -1,11 +1,34 @@
 library storage;
 
+/// Store a stack variable in storage.
 pub fn store<T>(key: b256, value: T) {
-    asm(r1: key, r2: value) {
-        sww r1 r2;
+    if !is_reference_type::<T>() {
+        // If copy type, then it's a single word and can be stored with a single SWW.
+        asm(r1: key, r2: value) {
+            sww r1 r2;
+        };
+    } else {
+        // If reference type, then it's more than a word. Loop over every 32 bytes and
+        // store sequentially.
+        let mut size_left = size_of::<T>();
+        let mut ptr = asm(r1, r2: value) {
+            move r1 r2;
+            r1
+        };
+        while size_left > 32 {
+            asm(r1: key, r2: ptr) {
```

Comment:
We're incrementing the pointer by `32` each cycle, but is there any issue with using the same `key` for each successive `swwq` ?
That is, are we overwriting whatever was previously stored using `key` ?

Reply:
Good catch

Reply:
Temporarily resolved via `sha256`.